BetaCameras/Xtion2/Xtion2.cs

```diff
@@ -0,0 +1,182 @@
+        }
+
+        protected override void DisconnectImpl()
+        {
+            OpenNICAPI.DestroyStream(_colorStream);
+            OpenNICAPI.DestroyStream(_depthStream);
+            OpenNICAPI.DestroyStream(_irStream);
+
+            OpenNICAPI.CloseDevice(_device);
+        }
+
+        protected override void UpdateImpl()
+        {
+            if(IsChannelActive(ChannelNames.Color))
+            {
+                OpenNICAPI.ReleaseFrame(_colorFrame);
+                _colorFrame = OpenNICAPI.ReadFrame(_colorStream);
+            }
+
+            if (IsChannelActive(ChannelNames.ZImage))
+            {
+                OpenNICAPI.ReleaseFrame(_depthFrame);
+                _depthFrame = OpenNICAPI.ReadFrame(_depthStream);
+            }
+
+            if (IsChannelActive(ChannelNames.Intensity))
+            {
+                OpenNICAPI.ReleaseFrame(_irFrame);
+                _irFrame = OpenNICAPI.ReadFrame(_irStream);
+            }
+        }
+
+        protected override CameraImage CalcChannelImpl(string channelName)
+        {
+            if (channelName == ChannelNames.Color)
+            {
+                return OpenNICAPI.FrameToColorImage(_colorFrame);
+            }
+            if (channelName == ChannelNames.ZImage)
+            {
+                return OpenNICAPI.FrameToFloatImage(_depthFrame);
+            }
+            if (channelName == ChannelNames.Intensity)
+            {
+                return OpenNICAPI.FrameToFloatImage(_irFrame);
+            }
+
+            log.Error("Unexpected ChannelName in CalcChannel().");
+            return null;
+        }
+
+        protected override void ActivateChannelImpl(String channelName)
+        {
+            if (channelName == ChannelNames.Color)
+            {
+                if (IsChannelActive(ChannelNames.Intensity))
+                {
+                    string msg = string.Format("Xtion2: Can't have {0} stream active while {1} is still active.", channelName, ChannelNames.Intensity);
+                    log.Error(msg);
+                    throw new InvalidOperationException(msg);
+                }  
+                
+                OpenNICAPI.StartStream(_colorStream);
+            }
+            else if (channelName == ChannelNames.ZImage)
+            {
+                OpenNICAPI.StartStream(_depthStream);
+            }
+            else if(channelName == ChannelNames.Intensity)
+            {
+                if (IsChannelActive(ChannelNames.Color))
+                {
+                    string msg = string.Format("Xtion2: Can't have {0} stream active while {1} is still active.", channelName, ChannelNames.Color);
+                    log.Error(msg);
+                    throw new InvalidOperationException(msg);
+                }
+
+                OpenNICAPI.StartStream(_irStream);
+            }
+            else
+            {
+                string msg = string.Format("Xtion2: doesn't implement channel.", channelName);
```

Comment:
Format string does not contain placeholder `{0}`

Reply:
Literal `Xtion2` should be replaced by `Name` property (everywhere).

Reply:
fixed